### DISCUSSION TITLE
column select

### DISCUSSION THREAD

**Discussion by clayton256** [orbitalquark/textadept#96] [Q&A] [answered]
Coming from years using VIM...

Is there a built-in way to select columns?

If not what would it take to add this?

**Comment by rgieseke** [comment, marked as answer]
It probably doesn't work exactly like colum selection in Vim, but Textadept has rectangular selection.

>You can create a rectangular selection as follows:
> 
>     Press the arrow keys, home/end, or page up/down, while holding down Alt+Shift on Windows, Linux, and BSD, ⌥⇧ on macOS, and M-S- in the terminal version.
>     Click and drag the mouse while holding down the Alt key on Windows, Linux, and BSD, and ⌥ on macOS.
>     Click and drag the mouse without holding down any modifiers (thus making a normal, multi-line selection), press and hold down the Alt key on Windows, Linux, and BSD, ⌥ on macOS, and M- in the terminal version, and then continue dragging the mouse. This feature is available because some window managers in Linux consume Alt+Shift + arrow keys and Alt + mouse drag.
> 
>

**Reply by clayton256** [reply]
Awesome! That's exactly what I needed. This looks like it was pasted from somewhere. Where is that so I can help myself next time.

**Reply by rgieseke** [reply]
Forgot to add the source link :-)
It's from the manual: https://orbitalquark.github.io/textadept/manual.html#text-selections

**Reply by rgieseke** [reply]
In case you haven't seen it, there have been a few attempts at bringing vi keybindings to Textadept, e.g. https://github.com/jugglerchris/textadept-vi from @jugglerchris

**Reply by jugglerchris** [reply]
Textadept-vi hasn't yet got rectangular selection as it's not something I've needed myself, but contributions and feature requests are welcome. :-)

**Reply by clayton256** [reply]
Thanks for the suggestion! I'm trying to get away from VI keybindings and move to CUA keybindings. I love how fast I can move around a file with vim but then my fingers try to do the same thing in libreoffice, mail, etc. 
Thanks for the link to the manual. I don't know how I missed it when pausing the website. Reading the manual answered my next question... how to delete a line.
I'm going to close this as answered. Re-open if there is anything else pertinent to add.
